kubeadm/v1.15.0/KubeClusterHelper.psm1

```diff
@@ -0,0 +1,1521 @@
+    }
+}
+
+function PrintConfig()
+{
+    ######################################################################################################################
+
+    Write-Host "############################################"
+    Write-Host "User Input "
+    Write-Host "Destination       : $Global:BaseDir"
+    Write-Host "Master            : $Global:MasterIp"
+    Write-Host "InterfaceName     : $Global:InterfaceName"
+    Write-Host "Cri               : $Global:Cri"
+    Write-Host "Cni               : $Global:Cni"
+    Write-Host "NetworkPlugin     : $Global:NetworkPlugin" 
+    Write-Host "Release           : $Global:Release"
+    Write-Host "MasterIp          : $Global:MasterIp"
+    Write-Host "ManagementIp      : $Global:ManagementIp"
+    Write-Host "ManagementSubnet  : $Global:ManagementSubnet"
+    Write-Host "############################################"
+
+    ######################################################################################################################
+}
+
+function Cleanup()
+{
+    
+}
+
+
+###################################################################################################
+
+function Expand-GZip($infile, $outfile = ($infile -replace '\.gz$',''))
+{
+    # From https://social.technet.microsoft.com/Forums/en-US/5aa53fef-5229-4313-a035-8b3a38ab93f5/unzip-gz-files-using-powershell?forum=winserverpowershell
+    $input = New-Object System.IO.FileStream $inFile, ([IO.FileMode]::Open), ([IO.FileAccess]::Read), ([IO.FileShare]::Read)
+    $output = New-Object System.IO.FileStream $outFile, ([IO.FileMode]::Create), ([IO.FileAccess]::Write), ([IO.FileShare]::None)
+    $gzipStream = New-Object System.IO.Compression.GzipStream $input, ([IO.Compression.CompressionMode]::Decompress)
+    try {
+        if (!$input -or !$output -or !$gzipStream)
+        {
+            throw "Failed to Unzip the archive"
+        }
+        $buffer = New-Object byte[](1024)
+        while($true){
+            $read = $gzipstream.Read($buffer, 0, 1024)
+            if ($read -le 0){break}
+            $output.Write($buffer, 0, $read)
+        }
+    } finally {
+        $gzipStream.Close()
+        $output.Close()
+        $input.Close()
+    }
+}
+
+function Install-7Zip()
+{
+    # ask to install 7zip, if it's not already installed
+    if (-not (Get-Command Expand-7Zip -ErrorAction Ignore)) {
+        Install-Package -Scope CurrentUser -Force 7Zip4PowerShell -Verbose
+        if(-not $?) {
+            Write-Error "Failed to install package"
+            Exit 1
+        }
+    }
+}
+
+function DownloadAndExtractTarGz($url, $dstPath)
+{
+    $tmpTarGz = New-TemporaryFile | Rename-Item -NewName { $_ -replace 'tmp$', 'tar.gz' } -PassThru
+    $tmpTar = New-TemporaryFile | Rename-Item -NewName { $_ -replace 'tmp$', 'tar' } -PassThru
+    DownloadFile -Url $url -Destination $tmpTarGz.FullName -Force
+    #Invoke-WebRequest $url -o $tmpTarGz.FullName
+    Expand-GZip $tmpTarGz.FullName $tmpTar.FullName
```

Comment:
Why not just use the BSD tar that is included with Windows? https://techcommunity.microsoft.com/t5/Containers/Tar-and-Curl-Come-to-Windows/ba-p/382409

Reply:
Done